File: day_1.R
library(dplyr)

inputs <- read.table(
  "Inputs/Input day 1",
  header = FALSE,
  col.names = c("var1", "var2")
)

solutions <- inputs %>%
  mutate(across(c(var1, var2),
                sort)) %>%
  left_join({.} %>% count(var2, name = "var2_count"),
            by = c("var1" = "var2")) %>%
  summarise(diff = sum(abs(var1 - var2)),
            sim = sum(var1 * var2_count, na.rm = TRUE))
  

File: day_1.py
list1 = []
list2 = []

with open("Inputs/Input day 1", "r") as input_file:
    for line in input_file:
        if(len(line) > 1):
            parts = line.split("   ")
            list1.append(int(parts[0]))
            list2.append(int(parts[1].rstrip('\n')))

#Part 1
list1.sort()
list2.sort()
            
diffs = [abs(x - y) for x, y in zip(list1, list2)]
solution = sum(diffs)

#Part 2
similarity = [i * list2.count(i) for i in list1]
solution2 = sum(similarity)
        

    
    

